replaced "divisor" with "delimiter" for methods split,rsplit and split_floats.

zakstam opened the pull request:
Supersedes #29430.
Fixes #29429.

aaronfranke commented:
This change should be placed along with your other changes in your previous PR.

Please see [this article](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html) for how to rebase, squash, and force push so that you just have one commit. I also created [this video](https://www.youtube.com/watch?v=0XKy8YpP5Ww) to show how it's done with a GUI client.

zaksnet commented:
> This change should be placed along with your other changes in your previous PR.
> Please see this article for how to rebase, squash, and force push so that you just have one commit. I also created this video to show how it's done with a GUI client.

Thank you Aaron, great video!

zakstam committed:
Improved documentation of rsplit Method for String class.

Improved documentation of rsplit Method for String class.

Removed "divisor" (i will also change variants_call.cpp) and added "delimiter" in its place. Also moved the example at the bottom of the description.

aaronfranke commented:
git rebase -i HEAD~7

Change all of your commits from "pick" to "fixup", except for the top one. Then:

    git push --force

zaksnet commented:
> ```
> git rebase -i HEAD~7
> ```
> 
> Change all of your commits from "pick" to "fixup", except for the top one. Then:
> 
> ```
> git push --force
> ```

I probably did something very wrong, i see commits from other users also:
![Screenshot_8](https://user-images.githubusercontent.com/1850856/59254774-82621000-8c39-11e9-99eb-33f94117cfba.png)

aaronfranke commented:
@zaksnet It doesn't look like your branch is checked out, since I *only* see other people's commits. If you wish to try a graphical client, this is easy to do in GitKraken:

![1](https://user-images.githubusercontent.com/1646875/59255670-6a21cf00-8c00-11e9-90f9-f9a053e110b4.png)

When you manage to do it correctly, on the top where it says "Commits" the number will be "1".

zaksnet commented:
> @zaksnet It doesn't look like your branch is checked out, since I _only_ see other people's commits. If you wish to try a graphical client, this is easy to do in GitKraken:
> 
> ![1](https://user-images.githubusercontent.com/1646875/59255670-6a21cf00-8c00-11e9-90f9-f9a053e110b4.png)
> 
> When you manage to do it correctly, on the top where it says "Commits" the number will be "1".

I think i found gitkrakken harder to use than cmd. It seems to be fine now, i removed the commits from other people and added fixup to my own commits. Anyway, i think i'm done with git for now, thanks for the help.

aaronfranke commented:
Awesome job, the git history looks good now. But it looks like you missed a few spots. The argument on line 665 is still "divisor" and the docs for `split` and `split_floats` on lines 710 to 736 need updating.

zaksnet commented:
> Awesome job, the git history looks good now. But it looks like you missed a few spots. The argument on line 665 is still "divisor" and the docs for `split` and `split_floats` on lines 710 to 736 need updating.

Thank you!

aaronfranke reviewed (approved):
The documentation changes and git history all look good now, and are good improvements.

Closes https://github.com/godotengine/godot/pull/29430, closes https://github.com/godotengine/godot/issues/29429, @akien-mga, @godotengine/documentation (this doesn't work?)

akien-mga commented on doc/classes/String.xml:
```diff
@@ -662,16 +662,17 @@
+				Splits the string by a [code]delimiter[/code] string and returns an array of the substrings, starting from right.
+				The splits in the returned array are sorted in the same order as the original string, from left to right.
+				If [code]maxsplit[/code] is specified, it defines the number of splits to do from the right up to [code]maxsplit[/code]. The default value of 0 means that all items are split, thus giving the same result as [method split].
+				[b]Example:[/b] [code]"One,Two,Three,Four"[/code] will return [code]["Three","Four"][/code] if split by [code]","[/code] with maxsplit of [code]"2"[/code].
```
As mentioned previously, the integer value `2` should not be within double quotes, which are the text representation of Strings. So it should be `with a [code]maxsplit[/code] value of 2`.

It would also be great if you could do similar changes to the `split` description for consistency.

Otherwise the changes are great, thanks a lot for putting up with the cumbersome Git workflow. You'll see that it gets easier :)

zaksnet replied:
> As mentioned previously, the integer value 2 should not be within double quotes, which are the text representation of Strings. So it should be with a [code]maxsplit[/code] value of 2.

So values should not be within [code][/code] tags? Should i also remove the tags from [code]","[/code] since its a value for the delimiter?

> It would also be great if you could do similar changes to the split description for consistency.

Sure!

> Otherwise the changes are great, thanks a lot for putting up with the cumbersome Git workflow. You'll see that it gets easier :)

Thank you! I think i'm getting the hang of it now that i switched to cmd. Managed to remove the commits from master (#29430) and close it since it was a duplicate to this one. I tend to make a mess when i learn and thats mostly ok if i'm by myself but now i felt i'm messing with other peoples work also.

aaronfranke replied:
Don't worry about messing up other people's work. The absolute worst case scenario is you mess up your copy of other people's work. But it seems that you have it figured out now.

akien-mga replied:
> So values should not be within [code][/code] tags? Should i also remove the tags from [code]","[/code] since its a value for the delimiter? Or just remove quotes from the integer value?

I don't have a strong opinion on that one, I guess you can use either `2` or `[code]2[/code]`. To me the latter emphasizes that it's the programming value "2" which represents an integer, while the former would be the "usual" number 2 that we refer to in plain English. (Don't know if that makes sense...)

akien-mga commented:
Great work, thanks a lot! :)

akien-mga commented:
Cherry-picked for 3.1.2.
